x-pack/plugins/apm/public/services/rest/apm/apm.ts

```diff
@@ -19,3 +25,54 @@ export async function getEncodedEsQuery(kuery?: string) {
   const esFilterQuery = convertKueryToEsQuery(kuery, indexPattern);
   return encodeURIComponent(JSON.stringify(esFilterQuery));
 }
+
+export async function getKueryESFilter(kuery?: string): Promise<ESFilter | undefined> {
+  if (!kuery) {
+    return;
+  }
+
+  const indexPattern = await getAPMIndexPatternForKuery();
+  if (!indexPattern) {
+    return;
+  }
+
+  return convertKueryToEsQuery(kuery, indexPattern) as ESFilter;
+}
+
+export function getEnvironmentESFilter(environment?: string): ESFilter | undefined {
+
+  if (!environment) {
+    return undefined;
+  }
+
+  if (environment === 'Null') {
+    return {
+      bool: { must_not: { exists: { field: 'service.environment' } } }
+    };
+  }
+  return {
+    term: { 'service.environment': environment }
+  };
+}
+
+export function getUIFilters(urlParams: IUrlParams): UIFilters {
+  return {
+    global: {
+      kuery: urlParams.kuery,
+      environment: urlParams.environment
+    }
+  };
+}
+
+export async function getEncodedEsFilters(uiFilters: UIFilters): Promise<string> {
+  const { global } = uiFilters;
+  const kuery = await getKueryESFilter(global.kuery);
```

Comment:
I just recall "global" is a reserved keyword in Node, like `window` is in the browser. Perhaps we don't need the separation between global and local filters?

Reply:
The only reason I wanted the separation was to make it easier to persist just the global but I think we can do that anyway. So `getUIFilters` should probably just return a flat object.

Reply:
i was wondering the same thing...the nesting seems arbitrary at this point, since the uiFilters object ultimate gets transforms into an array of filters anyway.